app/javascript/shared/components/useListNavigation.js

```diff
@@ -211,15 +211,29 @@ function getPreviousElement(
 }
 
 /**
- * Gets the first visible element that matches a selector
+ * Checks if the first completely visible element is present that matches a selector and returns if it is available
+ * If that isn't visible, it looks for the partially visible element that is present and returns that
+ * If no elements visible(like the banner case which could cover the entire viewport), we select the first element from the list
  *
  * @param {string} selector - The CSS selector
  *
  * @returns {object} The first visible element
  */
 function getFirstVisibleElement(selector) {
   const elements = document.querySelectorAll(selector);
-  return Array.prototype.find.call(elements, (element) =>
-    isInViewport(element),
+  let completelyVisibleFirstElement = Array.prototype.find.call(
```

Comment:
we don't change this so we can use `const`

```suggestion
  const completelyVisibleFirstElement = Array.prototype.find.call(
```

Reply:
Makes sense. Have addressed the same.